src/Logging/Logging.Testing/src/Xunit/DeflakeAttribute.cs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel;
+
+namespace Microsoft.Extensions.Logging.Testing
+{
+    /// <summary>
+    /// Runs a test multiple times to stress flaky tests that are believed to be fixed.
+    /// This can be used on an assembly, class, or method name.
+    /// Requires using <see cref="LoggedTest"/> to run.
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    [AttributeUsage(AttributeTargets.Method | AttributeTargets.Class | AttributeTargets.Assembly, AllowMultiple = false)]
+    public class DeflakeAttribute : Attribute
```

Comment:
Cute name, but let's be a little more descriptive :). Perhaps `RepeatAttribute`?

Reply:
We already have a `RetryAttribute` as a name, which will retry a test until it succeeds. I don't think `RepeatAttribute` is a great name as it's hard to know the difference between Repeat and Retry. But I don't have anything better...

Reply:
I think Repeat is sufficient to me. The difference between `Repeat` and `Retry` in my mind is clear. `Repeat` re-runs the test no matter what. `Retry` indicates there was a first "try" and it failed.

Reply:
🆙 📅